src/Orleans/Runtime/AsynchAgent.cs

```diff
@@ -161,8 +162,6 @@ private static void AgentThreadProc(Object obj)
             var agent = obj as AsynchAgent;
             if (agent == null)
             {
-                var log = LogManager.GetLogger("RuntimeCore.AsynchAgent");
-                log.Error(ErrorCode.Runtime_Error_100022, "Agent thread started with incorrect parameter type");
```

Comment:
Should this throw an exception or something? I don't believe this path is even possible, that's why I would throw an InvalidOperationException if we break that invariant at some point.

Reply:
I don't think this path is possible. That's why I removed the logging. But do you think it is worth to throw an exception here?

Reply:
I threw an exception